build-system/tasks/dep-check.js

```diff
@@ -130,24 +131,27 @@ Rule.prototype.matchBadDeps = function(moduleName, deps) {
           }
         }
 
-        const inWhitelist = this.whitelist_.some(entry => {
+        for (const entry of this.whitelist_) {
           const pair = entry.split('->');
           const whitelistedModuleName = pair[0];
           const whitelistedDep = pair[1];
           if (!minimatch(moduleName, whitelistedModuleName)) {
-            return false;
+            continue;
+          }
+
+          const inWhitelist = dep == whitelistedDep;
+          if (inWhitelist) {
```

Comment:
The result of line 142 is only used on line 143. Maybe remove the `const` and directly use the condition in line 143?

Reply:
Done